packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2646,7 +2650,12 @@ export class ContainerRuntime
 		this.channelCollection.setConnectionState(connected, clientId);
 		this.garbageCollector.setConnectionState(connected, clientId);
 
-		raiseConnectedEvent(this.mc.logger, this, connected, clientId);
+		// 'connected' here is a lie sent to the runtime that indicates whether the container
+		// can send ops or not (connected && !readonly). We can get the actual connection
+		// state from the container context to raise connected events properly.
+		const actualConnected = this._context.connected;
```

Comment:
How about we rename `connected` in this fn to `canSendOps`, and then this can be `connected`.  And your comment can go into the doc comment for this function explaining the interpretation of the fn parameter.

In fact, I would rename `_connected` on the class to `canSendOps` as well, and then on public `connected` getter, put a comment there too that this indicates whether the runtime is in a state where it can send ops.

Basically, find all usages of these and any related variables and think about whether they should be called `connected` or `canSendOps`.  Obviously don't change any of the public surface of the class though.

Somewhere that this function propagates

Reply:
Maybe you even change the implementation of `private canSendOps(): boolean`. I think you'd also want to change this snipped in the constructor too (to also check readonly), think it over:

```
		// Note that we only need to pull the *initial* connected state from the context.
		// Later updates come through calls to setConnectionState.
		this._connected = connected;
```